core/domain/prod_validation_jobs_one_off.py

```diff
@@ -2474,213 +2600,1326 @@ def _get_custom_validation_functions(cls):
+            feconf.DASHBOARD_STATS_DATETIME_STRING_FORMAT)
+        for stat in item.weekly_creator_stats_list:
+            for key, value in stat.iteritems():
+                is_invalid = False
+                try:
+                    datetime.datetime.strptime(
+                        key, feconf.DASHBOARD_STATS_DATETIME_STRING_FORMAT)
+                except Exception:
+                    is_invalid = True
+                if not is_invalid and key > current_time_str:
+                    is_invalid = True
+                if not isinstance(value, dict):
+                    is_invalid = True
+                allowed_properties = [
+                    'average_ratings', 'num_ratings', 'total_plays']
+                if not is_invalid and (
+                        sorted(value.keys()) != allowed_properties):
+                    is_invalid = True
+                if not is_invalid:
+                    for property_name in allowed_properties:
+                        if not isinstance(value[property_name], int):
+                            is_invalid = True
+                if is_invalid:
+                    cls.errors['weekly creator stats list'].append(
+                        'Entity id %s: Invalid stats dict: %s' % (
+                            item.id, stat))
+
+    @classmethod
+    def _get_custom_validation_functions(cls):
+        return [
+            cls._validate_schema_version,
+            cls._validate_weekly_creator_stats_list]
+
+
+class ExplorationUserDataModelValidator(BaseUserModelValidator):
+    """Class for validating ExplorationUserDataModels."""
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        return '^%s\\.%s$' % (item.user_id, item.exploration_id)
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'user_settings_ids': (
+                user_models.UserSettingsModel, [item.user_id]),
+            'exploration_ids': (
+                exp_models.ExplorationModel, [item.exploration_id])
+        }
+
+    @classmethod
+    def _validate_draft_change_list(cls, item):
+        """Validates that commands in draft change list follow
+        the schema of ExplorationChange domain object.
+
+        Args:
+            item: ndb.Model. ExplorationUserDataModel to validate.
+        """
+        for change_dict in item.draft_change_list:
+            try:
+                exp_domain.ExplorationChange(change_dict)
+            except Exception as e:
+                cls.errors['draft change list check'].append(
+                    'Entity id %s: Invalid change dict %s due to error %s' % (
+                        item.id, change_dict, e))
+
+    @classmethod
+    def _validate_rating(cls, item):
+        """Validates that rating is in the interval [1, 5].
+
+        Args:
+            item: ndb.Model. ExplorationUserDataModel to validate.
+        """
+        if item.rating and (item.rating < 1 or item.rating > 5):
```

Comment:
Are you allowing an item.rating of 0? If not, use `if item.rating is not None` rather than `if item.rating`.

Similarly elsewhere.

Reply:
Fixed.